test/framework/src/main/java/org/opensearch/test/OpenSearchIntegTestCase.java

```diff
@@ -1293,11 +1300,32 @@ protected final IndexResponse index(String index, String type, String id, String
     }
 
     /**
-     * Waits for relocations and refreshes all indices in the cluster.
+     * Waits for relocations and refreshes all indices in the cluster. Then waits for replicas
+     * to catch up with primary when segment replication is enabled.
      *
      * @see #waitForRelocation()
      */
     protected final RefreshResponse refresh(String... indices) {
```

Comment:
I know you asked me this offline yesterday I and gave you a different answer, but after looking at the code my opinion has changed :) ... I think `refresh()` should just probably stay the way it is, and a new method should be introduced called something like `void waitForReplication(String... indices)`. Internally it will call refresh and then wait for replicas to catch up. That means that all the places that call `refresh()` now but actually need to wait for replication will change, but it is probably better to be more explicit and we're not overloading the term "refresh" to mean something slightly different than what the actual refresh API means.